x-pack/plugins/observability/public/pages/slos/components/slo_list_search_filter_sort_bar.tsx

```diff
@@ -0,0 +1,192 @@
+  EuiFilterGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelectable,
+  EuiSelectableOption,
+} from '@elastic/eui';
+import { EuiSelectableOptionCheckedType } from '@elastic/eui/src/components/selectable/selectable_option';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+
+export interface SloListSearchFilterSortBarProps {
+  loading: boolean;
+  onChangeQuery: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onChangeSort: (sortMethod: SortType) => void;
+  onChangeStatusFilter: (statusFilters: SortItem[]) => void;
+}
+
+export type SortType = 'difference' | 'budgetRemaining';
+export type StatusType = 'violated' | 'forecastedViolation' | 'healthy';
+
+export type SortItem = EuiSelectableOption & {
+  label: string;
+  type: SortType | StatusType;
+  checked?: EuiSelectableOptionCheckedType;
+};
+
+const SORT_OPTIONS: SortItem[] = [
+  {
+    label: i18n.translate('xpack.observability.slos.list.sortBy.difference', {
+      defaultMessage: 'Difference',
+    }),
+    type: 'difference',
+    checked: 'on',
+  },
+  {
+    label: i18n.translate('xpack.observability.slos.list.sortBy.budgetRemaining', {
+      defaultMessage: 'Budget remaining',
+    }),
+    type: 'budgetRemaining',
+  },
+];
+
+const STATUS_OPTIONS: SortItem[] = [
+  {
+    label: i18n.translate('xpack.observability.slos.list.statusFilter.violated', {
+      defaultMessage: 'Violated',
+    }),
+    type: 'violated',
+  },
+  {
+    label: i18n.translate('xpack.observability.slos.list.statusFilter.healthy', {
+      defaultMessage: 'Healthy',
+    }),
+    type: 'healthy',
+  },
+];
+
+export function SloListSearchFilterSortBar({
+  loading,
+  onChangeQuery,
+  onChangeSort,
+  onChangeStatusFilter,
+}: SloListSearchFilterSortBarProps) {
+  const [isFilterPopoverOpen, setFilterPopoverOpen] = useState(false);
+  const [isSortPopoverOpen, setSortPopoverOpen] = useState(false);
+
+  const [sortOptions, setSortOptions] = useState(SORT_OPTIONS);
+  const [statusOptions, setStatusOptions] = useState(STATUS_OPTIONS);
+
+  const selectedSort = sortOptions.find((option) => option.checked === 'on');
+  const selectedStatusFilters = statusOptions.filter((option) => option.checked === 'on');
+
+  const handleToggleFilterButton = () => setFilterPopoverOpen(!isFilterPopoverOpen);
+  const handleToggleSortButton = () => setSortPopoverOpen(!isSortPopoverOpen);
+
+  const handleChangeSort = (newOptions: SortItem[]) => {
+    setSortOptions(newOptions);
+    setSortPopoverOpen(false);
+  };
+
+  const handleChangeStatusOptions = (newOptions: SortItem[]) => {
+    setStatusOptions(newOptions);
+    setFilterPopoverOpen(false);
+
+    onChangeStatusFilter(newOptions.filter((option) => option.checked === 'on'));
+  };
+
+  useEffect(() => {
+    if (selectedSort?.type === 'difference' || selectedSort?.type === 'budgetRemaining') {
+      onChangeSort(selectedSort.type);
+    }
+  }, [onChangeSort, selectedSort]);
```

Comment:
Can we move this into the `handleChangeSort` as we do for the `onChangeStatusFilter()` ?

Reply:
I'm afraid not. The component needs to pass the sorting method upwards upon mount. I could remove it but there is no guarantee that the BE will sort its results based on the same method that the UI reports (other than your word of course ;)